common/djangoapps/third_party_auth/models.py

```diff
@@ -0,0 +1,385 @@
+        default='{\n"SECURITY_CONFIG": {"metadataCacheDuration": 604800, "signMetadata": false}\n}',
+        help_text=(
+            "JSON object defining advanced settings that are passed on to python-saml. "
+            "Valid keys that can be set here include: SECURITY_CONFIG, SP_NAMEID_FORMATS, SP_EXTRA"
+        ),
+    )
+
+    class Meta(object):  # pylint: disable=missing-docstring
+        verbose_name = "SAML Configuration"
+        verbose_name_plural = verbose_name
+
+    def clean(self):
+        """ Standardize and validate fields """
+        super(SAMLConfiguration, self).clean()
+        self.org_info_str = clean_json(self.org_info_str, dict)
+        self.other_config_str = clean_json(self.other_config_str, dict)
+
+        self.private_key = self.private_key.replace("-----BEGIN PRIVATE KEY-----", "").strip()
+        self.private_key = self.private_key.replace("-----END PRIVATE KEY-----", "").strip()
+        self.public_key = self.public_key.replace("-----BEGIN CERTIFICATE-----", "").strip()
+        self.public_key = self.public_key.replace("-----END CERTIFICATE-----", "").strip()
+
+    def get_setting(self, name):
+        """ Get the value of a setting, or raise KeyError """
+        if name == "ORG_INFO":
+            return json.loads(self.org_info_str)
+        if name == "SP_ENTITY_ID":
+            return self.entity_id
+        if name == "SP_PUBLIC_CERT":
+            return self.public_key
+        if name == "SP_PRIVATE_KEY":
+            return self.private_key
+        if name == "TECHNICAL_CONTACT":
+            return {"givenName": "Technical Support", "emailAddress": settings.TECH_SUPPORT_EMAIL}
+        if name == "SUPPORT_CONTACT":
+            return {"givenName": "SAML Support", "emailAddress": settings.TECH_SUPPORT_EMAIL}
+        other_config = json.loads(self.other_config_str)
+        return other_config[name]  # SECURITY_CONFIG, SP_NAMEID_FORMATS, SP_EXTRA
+
+
+class SAMLProviderData(models.Model):
+    """
+    Data about a SAML IdP that is fetched automatically by 'manage.py saml pull'
+
+    This data is only required during the actual authentication process.
+    """
+    cache_timeout = 600
+    fetched_at = models.DateTimeField(db_index=True, null=False)
+    expires_at = models.DateTimeField(db_index=True, null=True)
+
+    entity_id = models.CharField(max_length=255, db_index=True)  # This is the key for lookups in this table
+    sso_url = models.URLField(verbose_name="SSO URL")
+    public_key = models.TextField()
+
+    class Meta(object):  # pylint: disable=missing-docstring
+        verbose_name = "SAML Provider Data"
+        verbose_name_plural = verbose_name
+        ordering = ('-fetched_at', )
+
+    def is_valid(self):
+        """ Is this data valid? """
+        if self.expires_at and timezone.now() > self.expires_at:
+            return False
+        return bool(self.entity_id and self.sso_url and self.public_key)
+    is_valid.boolean = True
```

Comment:
Yep. From [the `list_display` docs](https://docs.djangoproject.com/en/1.4/ref/contrib/admin/#django.contrib.admin.ModelAdmin.list_display): "If the string given is a method of the model, ModelAdmin or a callable that returns True or False Django will display a pretty “on” or “off” icon if you give the method a boolean attribute whose value is True."